File: lesson_5_2/app/src/main/java/com/kalashnyk/denys/samplemvvmapp/repository/database/dao/UserDao.java
package com.kalashnyk.denys.samplemvvmapp.repository.database.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import com.kalashnyk.denys.samplemvvmapp.repository.database.entity.UserEntity;

import java.util.List;

@Dao
public interface UserDao {
    @Query("SELECT * FROM users")
    List<UserEntity> getAll();

    @Query("SELECT * FROM users WHERE mId = :mId")
    UserEntity getById(int mId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertList(List<UserEntity> list);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void update(UserEntity userEntity);

    @Update
    void updateAll(List<UserEntity> list);

    @Delete
    void delete(UserEntity userEntity);
}

File: lesson_7/app/src/main/java/com/kalashnyk/denys/kotlinsample/repository/database/dao/UserDao.kt
package com.kalashnyk.denys.kotlinsample.repository.database.dao

import android.arch.persistence.room.*
import com.kalashnyk.denys.kotlinsample.repository.database.entity.UserEntity

@Dao
interface UserDao {
    @Query("SELECT * FROM users")
    fun getAll(): List<UserEntity>

    @Query("SELECT * FROM users WHERE id = :id")
    fun getById(id: Int): UserEntity

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertList(list: List<UserEntity>)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun update(userEntity: UserEntity)

    @Update
    fun updateAll(list: List<UserEntity>)

    @Delete
    fun delete(userEntity: UserEntity)
}